apiserver/pkg/http/client_test.go

```diff
@@ -1,17 +1,65 @@
 package http
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
 	"net/http"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 	rpcStatus "google.golang.org/genproto/googleapis/rpc/status"
 
+	"github.com/ray-project/kuberay/apiserver/pkg/util"
 	api "github.com/ray-project/kuberay/proto/go_client"
 )
 
+type mockTransport struct {
+	statusErr      *rpcStatus.Status
+	body           string
+	statusSequence []int
+	callCount      int
+	returnDoError  bool
+}
+
+func (m *mockTransport) RoundTrip(_ *http.Request) (*http.Response, error) {
```

Comment:
nit: add a documentation on behavior, especially when `m.statusSequence` all consumed, we return OK status.